src/NHibernate/Async/Impl/AbstractQueryImpl.cs

```diff
@@ -56,7 +56,5 @@ public abstract partial class AbstractQueryImpl : IQuery
 		}
 
 		#endregion
-
-		protected internal abstract Task<IEnumerable<ITranslator>> GetTranslatorsAsync(ISessionImplementor sessionImplementor, QueryParameters queryParameters, CancellationToken cancellationToken);
```

Comment:
Hmm... Adding to `AbstractQueryImpl`
```C#
// Since v5.2
[Obsolete("This method has no usages and will be removed in a future version")]
protected internal abstract Task<IEnumerable<ITranslator>> GetTranslatorsAsync(ISessionImplementor sessionImplementor, QueryParameters queryParameters, CancellationToken cancellationToken);
```
Leads async generator to generate non obsolete overrides in  `AbstractQueryImpl2` and `SqlQueryImpl` (even if manual obsolete override is added)
Should it be considered as async generator bug? Or did I miss something?

Reply:
For this case I think adding back manually the method in `AbstractQueryImpl2` non generated partial is the way to go.

Reply:
No. It's not recognized by async generator and duplicate method is added.

Reply:
With the version `0.8.2.7` both issues were corrected, so you can use one of the two mentioned approaches.